GDJS/GDJS/Extensions/Builtin/CommonInstructionsExtension.cpp

```diff
@@ -408,6 +409,87 @@ CommonInstructionsExtension::CommonInstructionsExtension() {
         return outputCode;
       });
 
+  GetAllEvents()["BuiltinCommonInstructions::ForEachStructure"]
+      .SetCodeGenerator([](gd::BaseEvent& event_,
+                           gd::EventsCodeGenerator& codeGenerator,
+                           gd::EventsCodeGenerationContext& parentContext) {
+        gd::String outputCode;
+        gd::ForEachStructureEvent& event =
+            dynamic_cast<gd::ForEachStructureEvent&>(event_);
+
+        // Context is "reset" each time the event is repeated (i.e. objects are
+        // picked again)
+        gd::EventsCodeGenerationContext context;
+        context.InheritsFrom(parentContext);
+        context.ForbidReuse();
+
+        gd::String conditionsCode = codeGenerator.GenerateConditionsListCode(
+            event.GetConditions(), context);
+        gd::String actionsCode =
+            codeGenerator.GenerateActionsListCode(event.GetActions(), context);
+        gd::String ifPredicat =
+            event.GetConditions().empty()
+                ? "true"
+                : codeGenerator.GenerateBooleanFullName(
+                      "condition" +
+                          gd::String::From(event.GetConditions().size() - 1) +
+                          "IsTrue",
+                      context) +
+                      ".val";
+
+        // Prepare object declaration and sub events
+        gd::String subevents =
+            codeGenerator.GenerateEventsListCode(event.GetSubEvents(), context);
+        gd::String objectDeclaration =
+            codeGenerator.GenerateObjectsDeclarationCode(context) + "\n";
+
+        // Write final code
+        gd::String iteratorReferenceVariableName =
+            "iteratorReference" +
+            gd::String::From(context.GetContextDepth());
+        gd::String structureChildVariableName =
+            "structureChild" +
+            gd::String::From(context.GetContextDepth());
+        gd::String structureReferenceVariableName =
+            "structureReference" +
+            gd::String::From(context.GetContextDepth());
```

Comment:
To keep things consistent with the changes proposed in the event, `structure` => `iterable`